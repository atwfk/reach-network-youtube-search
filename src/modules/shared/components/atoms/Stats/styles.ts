import styled from "styled-components";
import { COLORS, SCREENS } from "../../../constants";

export const StyledStats = styled.ul`
  display: block;
  color: ${COLORS.GREY_SHD1};
  font-weight: 400;
  font-size: 15px;

  @media (min-width: ${SCREENS.SM}) {
    display: flex;
  }

  li {
    margin-right: 6px;
    display: flex;
    align-items: center;

    .separator {
      margin-left: 6px;
      width: 2px;
      height: 2px;
      background-color: ${COLORS.GREY_SHD1};
      display: none;

      @media (min-width: ${SCREENS.SM}) {
        display: block;
      }
    }
  }
`;
