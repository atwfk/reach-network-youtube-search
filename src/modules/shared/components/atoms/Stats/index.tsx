import React from "react";
import type { FC, ReactElement } from "react";
import { StyledStats } from "./styles";

const Stats: FC<{ stats: string[] }> = ({ stats }): ReactElement => {
  return (
    <StyledStats className="stats">
      {stats.map((stat, index, items) => (
        <li key={stat} data-testid="stat">
          {stat} {items.length !== index + 1 && <span className="separator"></span>}
        </li>
      ))}
    </StyledStats>
  );
};

export default Stats;
