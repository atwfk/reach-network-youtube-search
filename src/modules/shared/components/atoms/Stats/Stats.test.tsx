import { render, screen, cleanup } from "@testing-library/react";
import "jest-styled-components";
import renderer from "react-test-renderer";
import Stats from ".";

afterEach(cleanup);

describe("Stats Component snapshots", () => {
  it("Should render three items of statistics with point separated", () => {
    const tree = renderer
      .create(<Stats stats={["TechWorld with Nana", "3.6M views", "a year ago"]} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("Should render only with item without point separated", () => {
    const tree = renderer.create(<Stats stats={["TechWorld with Nana"]} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe("Stats Component RTL", () => {
  it("should have three items of statistics", () => {
    render(<Stats stats={["TechWorld with Nana", "3.6M views", "a year ago"]} />);

    const stats = screen.getAllByTestId("stat");

    expect(stats).toHaveLength(3);
  });
});
