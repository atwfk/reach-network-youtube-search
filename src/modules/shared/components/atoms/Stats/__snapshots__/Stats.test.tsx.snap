// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stats Component snapshots Should render only with item without point separated 1`] = `
.c0 {
  display: block;
  color: #767272;
  font-weight: 400;
  font-size: 15px;
}

.c0 li {
  margin-right: 6px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 li .separator {
  margin-left: 6px;
  width: 2px;
  height: 2px;
  background-color: #767272;
  display: none;
}

@media (min-width:640px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media (min-width:640px) {
  .c0 li .separator {
    display: block;
  }
}

<ul
  className="c0 stats"
>
  <li
    data-testid="stat"
  >
    TechWorld with Nana
     
  </li>
</ul>
`;

exports[`Stats Component snapshots Should render three items of statistics with point separated 1`] = `
.c0 {
  display: block;
  color: #767272;
  font-weight: 400;
  font-size: 15px;
}

.c0 li {
  margin-right: 6px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 li .separator {
  margin-left: 6px;
  width: 2px;
  height: 2px;
  background-color: #767272;
  display: none;
}

@media (min-width:640px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media (min-width:640px) {
  .c0 li .separator {
    display: block;
  }
}

<ul
  className="c0 stats"
>
  <li
    data-testid="stat"
  >
    TechWorld with Nana
     
    <span
      className="separator"
    />
  </li>
  <li
    data-testid="stat"
  >
    3.6M views
     
    <span
      className="separator"
    />
  </li>
  <li
    data-testid="stat"
  >
    a year ago
     
  </li>
</ul>
`;
